test/jdk/security/infra/java/security/cert/CertPathValidator/certification/ValidatePathWithURL.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jtreg.SkippedException;
+
+import javax.net.ssl.*;
+import javax.security.auth.x500.X500Principal;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.security.*;
+import java.security.cert.*;
+import java.security.cert.Certificate;
+
+public class ValidatePathWithURL {
+
+    private final X509Certificate rootCertificate;
+    private final X500Principal rootPrincipal;
+
+    public ValidatePathWithURL(String caAlias) throws Exception {
+        System.setProperty("com.sun.net.ssl.checkRevocation", "true");
+        Security.setProperty("ssl.TrustManagerFactory.algorithm", "SunPKIX");
+
+        // some test sites don't have correct hostname specified in test certificate
+        HttpsURLConnection.setDefaultHostnameVerifier(new CustomHostnameVerifier());
+
+        String FS = System.getProperty("file.separator");
+        String CACERTS_STORE =
+                System.getProperty("test.jdk") + FS + "lib" + FS + "security" + FS + "cacerts";
+
+        KeyStore cacerts = KeyStore.getInstance("PKCS12");
+        try (FileInputStream fis = new FileInputStream(CACERTS_STORE)) {
+            cacerts.load(fis, null);
+        }
+
+        rootCertificate = (X509Certificate) cacerts.getCertificate(caAlias);
+        rootPrincipal = rootCertificate.getSubjectX500Principal();
+    }
+
+    public static void enableOCSPOnly() {
+        System.setProperty("com.sun.security.enableCRLDP", "false");
+        Security.setProperty("ocsp.enable", "true");
```

Comment:
Technically I think you could avoid setting these properties (and running in othervm mode) and instead init your `SSLContext` with a `TrustManager` and a `CertPathTrustManagerParameters` containing a `PKIXParameters` with a `PKIXRevocationChecker` configured with the desired revocation settings. However, AFAIK, you can't pass in an `SSLContext` via the `HttpsURLConnection` API, so you would need to use `HttpsClient` instead (which is only available in Java 11 and up). However, I think this is something to consider as a future enhancement, as you have more control over the various TLS settings.